### PR TITLE
add option to specify a proxied path prefix

### PR DESCRIPTION
**Pull request by peterneubauer**

### CODE REVIEW

**Comment by peterneubauer** [comment]
This is useful if you have mapped the ELS root path to something else than / (e.g. to provide auth). Would be great to get that into the next releas @aphyr !

File: bin/riemann-elasticsearch [hunk 1]
```diff
@@ -36,7 +37,7 @@ class Riemann::Tools::Elasticsearch
   end
 
   def health_url
-    "http://#{options[:es_host]}:#{options[:es_port]}/_cluster/health"
+    "http://#{options[:es_host]}:#{options[:es_port]}/#{options[:path_prefix]}/_cluster/health"
```
Comment (aphyr) [inline bin/riemann-elasticsearch]: Is a double slash (//) gonna work? Looks like the default is an empty string, so I want to make sure we don't break existing users.

Comment (peterneubauer) [inline bin/riemann-elasticsearch]: Normally they work, but better either test or parse it to single? Not sure
how to communicate the prefix format in a good way to the user...

Comment (aphyr) [inline bin/riemann-elasticsearch]: Ah, that too--I actually meant that by default, this is gonna hit `//_cluster/health`, not `/_cluster/health`.

Comment (peterneubauer) [inline bin/riemann-elasticsearch]: Yeah, feel free to remove one!

Comment (aphyr) [inline bin/riemann-elasticsearch]: Er, well, if we remove one, then... do we enforce path prefixes contain a trailing slash, or?

Comment (peterneubauer) [inline bin/riemann-elasticsearch]: Well,
We could remove any leading and trailing slash, and put it there ourselves,
so we get consistent behavior?

Comment (aphyr) [inline bin/riemann-elasticsearch]: Yep! Good call, but leave exactly one trailing slash if the prefix is non-empty, and drop one of the slashes from the URL. That way when the string is empty we get only one `/`, and when it's present, things work out correctly.

**Comment by aphyr** [comment]
Cool, thanks Peter! :D

**Pull request merged**